### PR TITLE
Added links to new features in usage document

### PR DESCRIPTION
**Pull request by imran-ansari**
* Copied documents related to new features from plugin_v2 to master branch
* Created links to new feature documents in usage document

@prablr79 , @wdurairaj - Please review

### CODE REVIEW

File: docs/create_snapshot_schedule.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+- -o expHrs=x                 This option is not mandatory option. x is an integer, indicates number of hours after which snapshot created
+                              via snapshot schedule will be deleted from 3PAR.
+- -o retHrs=x                 This option is not mandatory option. x is an integer, indicates number of hours this snapshot will be retained.
+
+docker command to create a snapshot schedule:
+```
+docker volume create -d hpe --name <snapshot_name> -o virtualCopyOf=volume1 
+-o scheduleFrequency="10 2 * * *" -o scheduleName=dailyOnceSchedule -o retentionHours=58 
+-o snaphotPrefix=pqr -o expHrs=5 -o retHrs=3
+```
```
[the remark this answers is not in the document]
Comment (imran-ansari) [reply docs/create_snapshot_schedule.md]: Fixed the typos

File: docs/create_snapshot_schedule.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+2. It creates a snapshot schedule on 3PAR with name for schedule as dailyOnceSchedule. 
+3. scheduleFrequency string specifies that task has to be created daily for each month and on each day at 10 minutes passed 2 O'clock.
+4. Docker snapshot has retentionHours of 58
+5. Snapshot created via scheduled snapshots will have prefix 'pqr' to its name and these snapshots will have retention period of 3 hours
+and expiration period of 5 hours
+
+###Inspect on volume and snapshot having a schedule associated with it.
+Consider volume1 is a volume for which snapshot schedule with name "ThisNewSnapSchedule" is created, for creating this schedule
+a Docker snapshot with name snapshot1 is created.
+
```
Comment (wdurairaj) [inline docs/create_snapshot_schedule.md]: You can add a note: 
If `-o scheduleName=auto` is passed via docker volume create , then the schedule name is generated based on a timestamp automatically. This is relevant for using the `scheduleName` in StorageClass in kubernetes environment.

File: docs/usage.md [hunk 3]
```diff
@@ -63,38 +70,73 @@ The **HPE 3PAR Docker Volume Plug-in** supports several optional parameters that
...
 - **qos-name** -- name of existing VVset on the HPE 3PAR where QoS rules are applied.(**introduced in plugin version 2.1**)
 
+- **replicationGroup** -- name of an existing remote copy group on the HPE 3PAR.(**introduced in plugin version 3.0**)
+
+- **fsOwner** -- user ID and group ID that should own root directory of file system.(**introduced in plugin version 3.0**)
+
+- **fsMode** -- mode of the root directory of file system to be specified as octal number.(**introduced in plugin version 3.0**)
+
+- **backend** -- backend to be used for the volume creation.(**introduced in plugin version 3.0**)
+
```
Comment (wdurairaj) [inline docs/usage.md]: Can we have a general help for getting the help options
`docker volume create -d hpe -o help` and
`docker volume create -d hpe -o help=backends`

File: docs/create_snapshot_schedule.md (hunk 2 above)
Comment (imran-ansari) [reply docs/create_snapshot_schedule.md]: Is it okay to add this as part of description of "scheduleName" itself?

- -o scheduleName=x           This option is mandatory. x is a string which indicates name for the schedule on 3PAR. If *scheduleName=auto* is passed via docker volume create , then the schedule name is generated automatically based on timestamp. This is relevant for using the scheduleName in StorageClass in Kubernetes environment.

File: docs/create_snapshot_schedule.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+                               hour          * or 0-23
+                               day-of-month  * or 1-31
+                               month         * or 1-12
+                               day-of-week   * or 0-6 (0 is Sunday)
+
+- -o scheduleName=x           This option is mandatory. x is a string which indicates name for the schedule on 3PAR.
+                              If *scheduleName=auto* is passed via docker volume create , then the schedule name is 
+                              generated automatically based on timestamp. This is relevant for using the scheduleName 
+                              in StorageClass in Kubernetes environment.
+- -o retentionHours=x         This option is not mandatory option. x is an integer, indicates number of hours this snapshot will be retained.
```
Comment (nilangekarss) [inline docs/create_snapshot_schedule.md]: retentionHours are not supported for creating a snapshot schedule. If this option is used while creating a snap schedule, error will be thrown.

File: docs/create_snapshot_schedule.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+- -o snapshotPrefix=x         This option is mandatory. x is prefix string for the scheduled snapshots which will get created on 3PAR
+- -o expHrs=x                 This option is not mandatory option. x is an integer, indicates number of hours after which snapshot created
+                              via snapshot schedule will be deleted from 3PAR.
+- -o retHrs=x                 This option is not mandatory option. x is an integer, indicates number of hours this snapshot will be retained.
+
+docker command to create a snapshot schedule:
+```
+$ docker volume create -d hpe --name <snapshot_name> -o virtualCopyOf=volume1 
+-o scheduleFrequency="10 2 * * *" -o scheduleName=dailyOnceSchedule -o retentionHours=58
```
Comment (nilangekarss) [inline docs/create_snapshot_schedule.md]: Please modify this command. retentionHours not supported while creating schedule.

File: docs/create_snapshot_schedule.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+$ docker volume create -d hpe --name <snapshot_name> -o virtualCopyOf=volume1 
+-o scheduleFrequency="10 2 * * *" -o scheduleName=dailyOnceSchedule -o retentionHours=58 
+-o snapshotPrefix=pqr -o expHrs=5 -o retHrs=3
+```
+
+#### Note:
+1. Above command creates a docker snapshot with name snapshot_name.
+2. It creates a snapshot schedule on 3PAR with name for schedule as dailyOnceSchedule. 
+3. scheduleFrequency string specifies that task has to be created daily for each month and on each day at 10 minutes passed 2 O'clock.
+4. Docker snapshot has retentionHours of 58
```
Comment (nilangekarss) [inline docs/create_snapshot_schedule.md]: Remove this point as command will not have this option.

File: docs/usage.md [hunk 7]
```diff
@@ -63,38 +73,76 @@ The **HPE 3PAR Docker Volume Plug-in** supports several optional parameters that
...
```
Comment (sonawane-shashikant) [inline docs/usage.md]: cpg is missing which is part of 3.0 release

File: docs/usage.md [hunk 8]
```diff
@@ -159,13 +207,30 @@ docker volume create -d hpe --name <snapshot_name> -o virtualCopyOf=<source_vol_
...
+$ docker volume create -d hpe -o help=backends
+```
+
+
+### Creating HPE 3PAR snapshot schedule(**introduced in plugin version 3.0**)
+```
+$ docker volume create -d hpe --name <snapshot-name> -o virtualCopyOf=<source-volume> 
+-o scheduleFrequency=<cron-format-string-within-double-quotes> -o scheduleName=<name> 
+-o snaphotPrefix=<snapshot-prefix> -o expHrs=<expiration-hours> -o retHrs=<retention-hours>
+```
```
Comment (sonawane-shashikant) [inline docs/usage.md]: snaphotPrefix is written. It should be snapshotPrefix.

**Pull request merged**